test/unit/errors.test.js

```diff
@@ -2,10 +2,31 @@
 
 const expect = require('chai').expect;
 const { getSymbolFrom } = require('../tools/utils');
-const MongoNetworkError = require('../../src/error').MongoNetworkError;
+const { MongoNetworkError } = require('../../src/index');
+const { PoolClosedError, WaitQueueTimeoutError } = require('../../src/cmap/errors');
 
-describe('MongoErrors', function () {
-  describe('MongoNetworkError', function () {
+describe('MongoErrors', () => {
+  // import errors as object
+  let errorClasses = Object.fromEntries(
+    Object.entries(require('../../src/index')).filter(([key]) => key.endsWith('Error'))
+  );
+  errorClasses = { ...errorClasses, PoolClosedError, WaitQueueTimeoutError };
+
+  for (const errorName in errorClasses) {
+    describe(errorName, () => {
+      it(`name should be read-only`, () => {
```

Comment:
technically, the code in this test does not check whether the error is readonly, but the particular implementation of the property as a getter; I usually advise to try to test the intention of the code, and not its implementation; in this case, it would mean taking the property, attempting to set it, and then expecting the property to be the same (something like `const obj = new TestClass()`, `expect(obj).to.have.property('name', 'Test Name')`, `obj.name = 'New Name'`, `expect(obj).to.have.property('name', 'Test Name')`

Reply:
Note: if you end up with a strict requirement for read only, the setting of the prop in the test should be a try/catch with an empty catch block (since that behavior is an implementation detail)

Reply:
Just updated the test to include testing the intention, thank you!